source/clear-linux/guides/maintenance/kernel-modules.rst

```diff
@@ -0,0 +1,186 @@
+signature verification of the |CL| kernel. For more information on |CL| 
+security practices, see the :ref:`security` page.
+
+
+Check if the module is already available through Clear Linux
+============================================================
+|CL| comes with many upstream kernel modules available for use.  If 
+you require a kernel module, be sure to check whether it is already available in |CL| first. 
+
+
+You can search for kernel modules using the :command:`swupd search` command. 
+See :ref:`swupd-search` for more information. 
+
+
+Consider requesting the module be added to |CL|
+===============================================
+If the kernel module you're needing is open source and likely to be useful to 
+others, consider submitting a request to add or enable to the |CL| kernel.
+
+You can make enhancement  requests to the |CL| distribution `on GitHub`_ .
+
+
+
+Building and loading out-of-tree modules
+========================================
+In some cases you may need an out-of-tree kernel module that is not able 
+to be made available through |CL|.
+
+.. note::
+
+   You can build and load out-of-tree kernel modules however secure boot must be 
+   disabled, kernel module integrity checking must be disabled, and you are 
+   responsible for building the module against new versions of the Linux kernel.
+   
+   Any time the kernel is upgraded on your Clear Linux system, you will 
+   need to rebuild your out-of-tree modules.
+
+
+This approach works well for individual development or testing. 
+For a more scalable and customizable approach, consider using the 
+`mixer tool`_ to provide a custom kernel and updates.
+
+
+Building kernel modules
+-----------------------
+
+#. From a |CL| system, ensure you are running the *native* kernel. 
+   Currently only the native kernel is enabled to build and load
+   out-of-tree modules.
+
+    .. code-block:: bash
+
+        $ uname -r
+        4.XX.YY-ZZZZ.native
+
+    Ensure *.native* is in the kernel name
+
+#. Install the `linux-dev` bundle to obtain the kernel headers, which are
+   required for compiling kernel modules
+
+    .. code-block:: bash
+
+        sudo swupd bundle-add linux-dev
+
+#. Follow instructions from the kernel module source code to compile the 
+   kernel module
+
+
+Loading kernel modules
+----------------------
+
+#. Disable Secure Boot in your BIOS or EFI you have enabled it. The loading of 
+   new out-of-tree modules modifies the signatures  Secure Boot relies on for 
+   trust. 
+
+
+#. Disable signature checking for the kernel by modifying the kernel boot 
+   parameters and reboot the system. 
+
+   All kernel modules from |CL| have been signed to enforce kernel security. 
+   However, out-of-tree modules break this chain of trust so this mechanism 
+   needs to be disabled.
+  
+    .. code-block:: bash
+
+        sudo mkdir -p /etc/kernel/cmdline.d
+        sudo echo "module.sig_unenforce" > /etc/kernel/cmdline.d/load-modules.conf
```

Comment:
sudo is useless here, as it applies only to the echo and not the redirect. Could rewrite as:
echo "module.sig_unenforce" | sudo tee /etc/kernel/cmdline.d/load-modules.conf

Reply:
Might also rename the example conf file to "allow-unsigned-modules.conf" to be more explicit